tests/build/build_templates.go

```diff
@@ -163,48 +164,63 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
-				Expect(err).ToNot(HaveOccurred())
-				err = kubeadminClient.CommonController.LinkSecretToServiceAccount(testNamespace, resultSecret.Name, resultSA, false)
-				Expect(err).ToNot(HaveOccurred())
+					resultSecret := &v1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:        "tekton-results-tests",
+							Annotations: map[string]string{"kubernetes.io/service-account.name": resultSA},
+						},
+						Type: v1.SecretTypeServiceAccountToken,
+					}
 
-				resultSecret, err = kubeadminClient.CommonController.GetSecret(testNamespace, resultSecret.Name)
-				Expect(err).ToNot(HaveOccurred())
-				token := resultSecret.Data["token"]
+					_, err = kubeadminClient.CommonController.CreateSecret(testNamespace, resultSecret)
+					Expect(err).ToNot(HaveOccurred())
+					err = kubeadminClient.CommonController.LinkSecretToServiceAccount(testNamespace, resultSecret.Name, resultSA, false)
+					Expect(err).ToNot(HaveOccurred())
 
-				// Retrive Result REST API url
-				resultRoute, err := kubeadminClient.CommonController.GetOpenshiftRoute("tekton-results", "tekton-results")
-				Expect(err).NotTo(HaveOccurred())
-				resultUrl := fmt.Sprintf("https://%s", resultRoute.Spec.Host)
-				resultClient := pipeline.NewClient(resultUrl, string(token))
+					resultSecret, err = kubeadminClient.CommonController.GetSecret(testNamespace, resultSecret.Name)
+					Expect(err).ToNot(HaveOccurred())
+					token := resultSecret.Data["token"]
 
-				pipelineRun, err := kubeadminClient.HasController.GetComponentPipelineRun(componentNames[i], applicationName, testNamespace, "")
-				Expect(err).ShouldNot(HaveOccurred())
+					// Retrive Result REST API url
+					resultRoute, err := kubeadminClient.CommonController.GetOpenshiftRoute("tekton-results", "tekton-results")
+					Expect(err).NotTo(HaveOccurred())
+					resultUrl := fmt.Sprintf("https://%s", resultRoute.Spec.Host)
+					resultClient = pipeline.NewClient(resultUrl, string(token))
 
-				// Verify Records
-				records, err := resultClient.GetRecords(testNamespace, string(pipelineRun.GetUID()))
-				Expect(err).NotTo(HaveOccurred())
-				Expect(len(records.Record)).NotTo(BeZero())
-				// Verify Logs
-				logs, err := resultClient.GetLogs(testNamespace, string(pipelineRun.GetUID()))
-				Expect(err).NotTo(HaveOccurred())
-				Expect(len(logs.Record)).NotTo(BeZero())
+					pipelineRun, err = kubeadminClient.HasController.GetComponentPipelineRun(componentNames[i], applicationName, testNamespace, "")
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				It("should have Pipeline Records", func() {
+					// Verify Records
+					records, err := resultClient.GetRecords(testNamespace, string(pipelineRun.GetUID()))
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(records.Record)).NotTo(BeZero(), "No records found for PipelineRun %s", pipelineRun.Name)
+				})
+
+				It("should have Pipeline Logs", func() {
+					// Verify Logs
+					// Verify if result is stored in Database
```

Comment:
Do we need this comment twice in the single `It` spec?

Reply:
fixed.